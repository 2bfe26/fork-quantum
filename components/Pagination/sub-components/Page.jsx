import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Colors from '../../Colors';

const PageButton = styled.a`
  align-items: center;
  background-color: ${({ active }) =>
    active ? Colors.BLUE['500'] : Colors.WHITE};
  border: 1px solid
    ${({ active }) => (active ? Colors.BLUE['500'] : Colors.BLACK['200'])};
  border-radius: 10px;
  box-sizing: border-box;
  cursor: pointer;
  display: flex;
  font-size: 16px;
  font-weight: ${({ active }) => (active ? '700' : 'normal')};
  justify-content: center;
  margin: 0 4px;
  min-height: 40px;
  min-width: 40px;
  outline: none;
  padding: 5px 8px;
  text-decoration: none;
  transition: background-color 0.2s ease-in-out, border-color 0.2s ease-in-out,
    color 0.2s ease-in-out, font-weight 0.2s ease-in-out;

  ${({ active }) => active && `color: ${Colors.WHITE};`};

  :hover,
  :focus {
    background-color: ${({ active }) =>
      active ? Colors.COBALT['500'] : Colors.BLUE['200']};
    border-color: ${({ active }) =>
      active ? Colors.COBALT['500'] : Colors.BLUE['500']};
    color: ${({ active }) => (active ? Colors.WHITE : Colors.BLUE['500'])};
  }
`;

const Page = ({ children, onClick, ...props }) => (
  <PageButton onClick={onClick} {...props}>
    {children}
  </PageButton>
);

Page.displayName = 'Page';

Page.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.number.isRequired,
  onClick: PropTypes.func,
};

Page.defaultProps = {
  active: false,
  onClick: () => {},
};

export default Page;
