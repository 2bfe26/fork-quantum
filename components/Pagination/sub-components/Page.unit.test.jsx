import React from 'react';
import renderer from 'react-test-renderer';
import Page from './Page';

describe('<Page />', () => {
  describe('Snapshot', () => {
    it('should match snapshot', () => {
      expect(renderer.create(<Page>{50}</Page>).toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with active prop', () => {
      expect(
        renderer.create(<Page active>{10}</Page>).toJSON(),
      ).toMatchSnapshot();
    });
  });
});
