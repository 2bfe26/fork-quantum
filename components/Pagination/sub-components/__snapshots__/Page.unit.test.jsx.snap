// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> Snapshot should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #cccccc;
  border-radius: 10px;
  box-sizing: border-box;
  color: #4c4c4c;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: normal;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 4px;
  min-height: 40px;
  min-width: 40px;
  outline: none;
  padding: 5px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out,color 0.2s ease-in-out,font-weight 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out,color 0.2s ease-in-out,font-weight 0.2s ease-in-out;
}

.c0:hover,
.c0:focus {
  background-color: #e5edfc;
  border-color: #1355d0;
  color: #1355d0;
}

<a
  className="c0"
  onClick={[Function]}
>
  50
</a>
`;
